Sources/Ollama/Value.swift

```diff
@@ -242,3 +242,22 @@ extension Value: ExpressibleByStringInterpolation {
         self = .string(stringInterpolation.stringValue)
     }
 }
+
+// MARK: - JSON Helper
+
+extension Value {
+    /// Convert the value to a Double if it is representable as a double,
+    /// otherwise nil
+    public var asDoubleValue: Double? {
```

Comment:
Value appears to already support this functionality as `[doubleValue](https://github.com/mattt/ollama-swift/blob/36839e1ce0e223a28b9c5c40cb99ce454604e3dd/Sources/Ollama/Value.swift#L49)`

Reply:
`doubleValue` is only if the parameter in question was parsed by the json parser as a floating point number. If it came through as an integer value, it'll be passed as an int. I have also seen LLMs tend to put numbers in String's not-uncommonly.

`doubleValue` is double if the value actually is double. It's what the value actually was.  
`asDoubleValue` Converts the value to a Double if it is representable as a double. Regardless of what it was in the data that came in.

I added it because writing my tests I found I had to sprinkle in

```swift
let actualValue: Double? = (value.double ?? value.int.flatMap({ Double($0) }) )
```

Every time I wanted to use a value. I couldn't just use `value.doubleValue` because if it came in without the `.0` it'd be parsed as an int and be `nil`.

Having that in every single one of my test functions (as all were math ones) made me realise, if I was going to use this library I'd need to just write an extension method to handle that for me or it'd get too annoying.  
So I just added it to the project, instead of having an extension method I just added to every single project I ever included this library in.

Reply:
I agree that flexible conversion from `Value` types to standard library types is a nice quality-of-life improvement. Personally, I think I prefer spelling this more like `Double.init?(_ value: Value, strict: Bool = true)`, [following Pydantic's lead](https://docs.pydantic.dev/dev/concepts/conversion_table/).

Reply:
I understand the thought process behind this, but I have a strong distaste for global extensions on standard types, like String, Double, Int, etc…

Especially if they leak into a client apps namespace & pollute its functions.

If every library added a few custom init methods. Then suddenly init Double would jump from having a few to dozens of non-stand extra methods all special cased to those libraries.

I generally don’t feel my code is /important/ enough to spread out across a client app beyond my API namespace into the global namespace.

As an API customer these helper methods will have to be used by every tool function every time. And `value.convertToDouble()` is much more ergonomic than `value?.flatMap(Double($0))` every time.